File: exercise/java/day10/src/test/java/DeliveryTest.java
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

import java.io.IOException;
import java.net.URISyntaxException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.Objects;

import static delivery.Building.whichFloor;
import static org.assertj.core.api.Assertions.assertThat;

class DeliveryTest {
    @ParameterizedTest
    @CsvSource({
            "1, 0",
            "2, 3",
            "3, -1",
            "4, 53",
            "5, -3",
            "6, 2920"
    })
    void returnFloorNumberBasedOnInstructions(String fileName, int expectedFloor) throws IOException, URISyntaxException {
        var instructions = loadInstructionsFromFile(fileName + ".txt");
        assertThat(whichFloor(instructions))
                .isEqualTo(expectedFloor);
    }

    private static String loadInstructionsFromFile(String input) throws URISyntaxException, IOException {
        return Files.readString(
                Path.of(Objects.requireNonNull(DeliveryTest.class.getClassLoader().getResource(input)).toURI())
        );
    }
}

File: exercise/kotlin/day10/src/test/kotlin/DeliveryTest.kt
import delivery.Building
import io.kotest.core.spec.style.StringSpec
import io.kotest.data.forAll
import io.kotest.data.row
import io.kotest.matchers.shouldBe
import java.nio.file.Files
import java.nio.file.Path
import java.util.*

fun loadInstructionsFromFile(input: String): String = Files.readString(
    Path.of(Objects.requireNonNull(DeliveryShould::class.java.getClassLoader().getResource(input)).toURI())
)

class DeliveryShould : StringSpec({
    "return floor number based on instructions" {
        forAll(
            row("1", 0),
            row("2", 3),
            row("3", -1),
            row("4", 53),
            row("5", -3),
            row("6", 2920)
        ) { fileName, expectedFloor ->
            loadInstructionsFromFile("$fileName.txt")
                .let { instructions ->
                    Building.whichFloor(instructions) shouldBe expectedFloor
                }
        }
    }
})